### PR TITLE
Fixes #2315 Empty Sidebar First region rendered on pages not in the menu

### LINKED ISSUE #2315: Empty Sidebar First region rendered on pages not in the menu
## Problem/Motivation
While testing Quickstart 2.6.1 on the [Test environment for the Think Tank site](https://test-azs-thinktank.pantheonsite.io/), we noticed that the sidebar region is appearing on all pages of the test site, whether those pages are in the sidebar menu or not.

### Describe the bug
In this case of the Think Tank site, the sidebar is using a custom menu instead of Main Navigation. For pages which are not in the sidebar menu, the HTML for the sidebar region is included but is empty:

```html
<div class="sidebar_first sidebar col-md-3 order-first" id="sidebar_first">
  <aside class="section" role="complementary">
                  
  </aside>
</div>
```

## Proposed resolution
This problem may be related to this Drupal issue: [Empty blocks get displayed](https://www.drupal.org/project/menu_block/issues/3280764). We could try applying the patch from this issue to fix the problem with the empty Sidebar First region.

However, this patch cannot be applied currently because recently we added a patch from another Drupal issue to Quickstart: [Menu block renders when tree is empty as of 8.x-1.8 [regression]](https://www.drupal.org/project/menu_block/issues/3271218).

@trackleft noted that patch 17 from this issue removes the use statement for AccountInterface:
`-use Drupal\Core\Session\AccountInterface;`

Since the patch from the first linked issue requires the use of AccountInterface for menu blocks, @trackleft suggested removing the other patch from Quickstart so that the "Empty blocks get displayed" patch could be applied.

### Expected behavior

The sidebar region should not appear if the region is empty (no sidebar menu blocks are displayed).

##  Screenshots
image

### PR DESCRIPTION
**Pull request by bberndt-uaz**
## Description

Replaces the patch from [this Menu Block issue](https://www.drupal.org/project/menu_block/issues/3271218) with a patch from a [different Menu Block issue](https://www.drupal.org/project/menu_block/issues/3280764), which appears to resolve a problem where the Sidebar First region is rendered even when it is empty.

## Related issues

#2315

## How to test

Test for the Sidebar First region on pages in the navigation menu and pages that are not in the navigation menu.

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [ ] Update experimental module
- **Minor release changes**
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [x] Patch level release (non-security bug-fix release)
   - [x] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -180,8 +180,8 @@
             "drupal/media_migration": {
                 "Catch migrate RequirementsExceptions (3268480)": "https://www.drupal.org/files/issues/2022-03-08/3268480-2.patch"
             },
-            "drupal/menu_block": {
-                "Menu block renders when tree is empty as of 8.x-1.8 [regression] (3271218)": "https://www.drupal.org/files/issues/2022-04-29/menu_block_rendered_empty-3271218-17.patch"
+             "drupal/menu_block": {
```
Comment (bberndt-uaz) [inline composer.json:183]:
```suggestion
            "drupal/menu_block": {
```

**Comment by trackleft** [comment]
Can probably get rid of this as well https://github.com/az-digital/az_quickstart/blob/main/modules/custom/az_core/az_core.module#L169-L181

File: modules/custom/az_core/az_core.module [hunk 2]
```diff
@@ -166,20 +166,6 @@ function az_core_modules_uninstalled($modules) {
...
-  if ($operation === 'view' && $block->id() === 'az_barrio_sidebar_menu') {
-    $block_content = \Drupal::entityTypeManager()
-      ->getViewBuilder('block')
-      ->view($block);
-    $markup = \Drupal::service('renderer')->renderRoot($block_content);
-    return AccessResult::forbiddenIf($markup === '')->addCacheableDependency($block);
-  }
-  return AccessResult::neutral();
-}
-
```
Comment (joeparsons) [inline modules/custom/az_core/az_core.module:182]: I think we should try @trackleft's suggestion to restore this hook and have it apply to all menu blocks, not just our specific menu block id:

```
/**
 * Implements hook_block_access().
 */
function az_core_block_access(Block $block, $operation, AccountInterface $account) {
  if ($operation === 'view' && $block->plugin() === 'menu_block') {
    $block_content = \Drupal::entityTypeManager()
      ->getViewBuilder('block')
      ->view($block);
    $markup = \Drupal::service('renderer')->renderRoot($block_content);
    return AccessResult::forbiddenIf($markup === '')->addCacheableDependency($block);
  }
  return AccessResult::neutral();
}
```

Comment (trackleft) [reply modules/custom/az_core/az_core.module:182]: I think menu_block and az_menu_block should be added

Comment (trackleft) [reply modules/custom/az_core/az_core.module:182]: Actually, something like this probably 
https://github.com/az-digital/az_quickstart/blob/main/modules/custom/az_block_types/az_custom_menu_block/az_custom_menu_block.module#L24-L25

File: modules/custom/az_core/az_core.module [hunk 3]
```diff
@@ -170,7 +170,7 @@ function az_core_modules_uninstalled($modules) {
  * Implements hook_block_access().
  */
 function az_core_block_access(Block $block, $operation, AccountInterface $account) {
-  if ($operation === 'view' && $block->id() === 'az_barrio_sidebar_menu') {
+  if ($operation === 'view' && $block->getPluginId() === 'menu_block') {
```
Comment (trackleft) [inline modules/custom/az_core/az_core.module:173]:
```suggestion
  if ($operation === 'view' && strpos($block->getPluginId(), 'menu_block:') === 0) {
```

**Pull request closed without merging**